### PR TITLE
Security manager API javadoc improvements

### PR DESCRIPTION
**Pull request by emanoelxavier**
1. Gerating javadoc as part of the build
2. Fixing javadoc build issues
3. Improving content
5. Improving text format in the code and javadoc artifact: consistent indentation, consistent sentences, use of paragraphs and line items, etc

### CODE REVIEW

**Comment by emanoelxavier** [comment]
To see the end result javadoc you look here https://ci.opensecuritycontroller.org:8081/nexus/content/repositories/osc-snapshot/org/osc/api/security-mgr-api/1.0.1-SNAPSHOT/security-mgr-api-1.0.1-20170707.150837-4-javadoc.jar

File: src/main/java/org/osc/sdk/manager/api/ManagerDeviceApi.java [hunk 1]
```diff
@@ -26,224 +26,190 @@
...
-     *            The member device id to delete
-     * @throws Exception
+     * @param id the identifier of the device to be deleted
+     * @throws Exception upon failure
      */
     void deleteDeviceMember(String id) throws Exception;
 
     /**
-     * Find member device using id from current VS device container
+     * Finds the device member given its identifier and the context of the current virtual system.
```
Comment (larkinscarvalho) [inline src/main/java/org/osc/sdk/manager/api/ManagerDeviceApi.java]: "Retrieves the device member..." to be consistent

File: src/main/java/org/osc/sdk/manager/api/ManagerDeviceApi.java [hunk 2]
```diff
@@ -26,224 +26,190 @@
...
-     * @throws Exception
+     * @param id the identifier of the targeted device member
+     * @return The device member, null if not found
+     * @throws Exception upon failure
      */
     ManagerDeviceMemberElement getDeviceMemberById(String id) throws Exception;
 
     /**
-     * Locate member device by name within the members of current VS device container
+     * Finds the device member given its name and the context of the current virtual system.
```
Comment (larkinscarvalho) [inline src/main/java/org/osc/sdk/manager/api/ManagerDeviceApi.java]: "Retrieves the device member..." to be consistent

File: src/main/java/org/osc/sdk/manager/api/ManagerDeviceApi.java [hunk 3]
```diff
@@ -26,224 +26,190 @@
...
+     * @param newSwVersion  the software version to upgrade to
+     * @return true if upgrade is supported
+     * @throws Exception upon failure
      */
     boolean isUpgradeSupported(String modelType, String prevSwVersion, String newSwVersion)
             throws Exception;
 
     /**
-     * Get individual member device configuration from Manager to be persisted in ISC. Returned blob will be downloaded
+     * Gets the configuration for a given device member. The returned blob will be downloaded
```
Comment (larkinscarvalho) [inline src/main/java/org/osc/sdk/manager/api/ManagerDeviceApi.java]: "Retrieves the device member..." to be consistent.

get/find: Can it be as retrieve operation? Multiple places

File: src/main/java/org/osc/sdk/manager/ManagerNotificationSubscriptionType.java [hunk 4]
```diff
@@ -29,10 +43,9 @@
     }
 
     /**
-     * Gets the {@link ManagerNotificationSubscriptionType} that corresponds to the provided string.
-     * @param test The string value of the targeted type.
-     * @return
-     *  Returns the type for the provided string.
+     * Gets the {@link ManagerNotificationSubscriptionType} corresponding to the provided string.
```
Comment (emanoelxavier) [inline src/main/java/org/osc/sdk/manager/ManagerNotificationSubscriptionType.java]: changed to 'provided string value'

File: src/main/java/org/osc/sdk/manager/api/ApplianceManagerApi.java [hunk 5]
```diff
@@ -16,128 +16,134 @@
...
-     *            Manager Connector Element {@link ApplianceManagerConnectorElement}
-     * @return Callback Notification Api
-     * @throws UnsupportedOperationException
-     *             if Callback Notification is not supported
-     * @throws Exception
+     * @param mc  provides information needed to connect with the security manager
+     * @return the APIs used by OSC to manage notification registration
+     *
+     * @throws UnsupportedOperationException if the notification type is
+     * not {@link ManagerNotificationSubscriptionType#CALLBACK_URL}
```
Comment (emanoelxavier) [inline src/main/java/org/osc/sdk/manager/api/ApplianceManagerApi.java]: In the javadoc it shows "notification type is not ManagerNotificationSubscriptionType.CALLBACK_URL" adding the suggested extra text will make it look redundant.

**Pull request merged**